js/binance.js

```diff
@@ -3582,9 +3611,24 @@ module.exports = class binance extends Exchange {
         if ((marketType === 'spot') || (marketType === 'margin')) {
             request['newOrderRespType'] = this.safeValue (this.options['newOrderRespType'], type, 'RESULT'); // 'ACK' for order id, 'RESULT' for full order or 'FULL' for order with fills
         } else {
-            // delivery and future
+            // swap, futures and options
             request['newOrderRespType'] = 'RESULT';  // "ACK", "RESULT", default "ACK"
         }
+        if (market['option']) {
+            if (type === 'market') {
+                throw new InvalidOrder (this.id + ' ' + type + ' is not a valid order type for the ' + symbol + ' market');
+            }
+            method = 'eapiPrivatePostOrder';
```

Comment:
@Dan-krm can you please try to place an options order using a RSA key? 
For some reason, I can place spot/futures orders normally, but options don't go through (and I have the enable options permission)

```
ResponseBody:
 {"code":-2015,"msg":"Invalid API-key, IP, or permissions for action, request ip: 2.80.182.225"} 

AuthenticationError binance {"code":-2015,"msg":"Invalid API-key, IP, or permissions for action, request ip: 2.80.182.225"}
---------------------------------------------------
[AuthenticationError] binance {"code":-2015,"msg":"Invalid API-key, IP, or permissions for action, request ip: 2.80.182.225"}

    at throwExactlyMatchedException  js/base/Exchange.js:2419              throw new exact[string] (message);                                              
    at handleErrors                  js/binance.js:6754                    this.throwExactlyMatchedException (this.exceptions['exact'], error, feedback);  
    at                               js/base/Exchange.js:603               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                  
    at async timeout                 js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                        js/base/Exchange.js:2042              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                       js/binance.js:6783                    const response = await this.fetch2 (path, api, method, params, headers, body, c…
    at createOrder                   js/binance.js:3747                    const response = await this[method] (this.extend (request, requestParams));     
    at async run                     examples/js/cli.js:293                const result = await exchange[methodName] (... args)                            

binance {"code":-2015,"msg":"Invalid API-key, IP, or permissions for action, request ip: 2.80.182.225"}
```

Reply:
@carlosmiei I'm not able to recreate the error, I was able to successfully place an options order using the RSA key approach. When I created the RSA key I had to recheck all the permissions after creating the key and I needed to use the generated private key as the secret including `-----BEGIN PRIVATE KEY-----` and ending with `-----END PRIVATE KEY-----`
I'm not sure what it could be, maybe they reset the permissions for the RSA key you're using during the system upgrade or something

They had a system upgrade on the 9th:
https://www.binance.com/en/support/announcement/binance-api-key-system-upgrade-notice-2023-02-09-a6274f9d2e0845068eaff7ef21afd793

Reply:
@Dan-krm ok I've created the keys but now I get a different issue 🤔 
```
n binance createOrder "ETH/USDT:USDT-230215-1450-C" limit buy 1 10 
Debugger attached.
2023-02-14T09:50:05.508Z
Node.js: v17.9.0
CCXT v2.7.78
binance.createOrder (ETH/USDT:USDT-230215-1450-C, limit, buy, 1, 10)
BadRequest binance {"code":-1130,"msg":"Invalid data sent for a parameter."}
---------------------------------------------------
[BadRequest] binance {"code":-1130,"msg":"Invalid data sent for a parameter."}

    at throwExactlyMatchedException  js/base/Exchange.js:2419              throw new exact[string] (message);                                              
    at handleErrors                  js/binance.js:6754                    this.throwExactlyMatchedException (this.exceptions['exact'], error, feedback);  
    at                               js/base/Exchange.js:603               const skipFurtherErrorHandling = this.handleErrors (response.status, response.s…
    at processTicksAndRejections     node:internal/process/task_queues:96                                                                                  
    at async timeout                 js/base/functions/time.js:181         return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at fetch2                        js/base/Exchange.js:2042              return await this.fetch (request['url'], request['method'], request['headers'],…
    at request                       js/binance.js:6783                    const response = await this.fetch2 (path, api, method, params, headers, body, c…
    at createOrder                   js/binance.js:3747                    const response = await this[method] (this.extend (request, requestParams));     
    at async run                     examples/js/cli.js:293                const result = await exchange[methodName] (... args)                            

binance {"code":-1130,"msg":"Invalid data sent for a parameter."}
```

Reply:
Sorry to see you're having these issues, I'm not able to pinpoint this one either, it's still working for me:
```
binance.createOrder (ETH/USDT:USDT-230215-1450-C, limit, buy, 1, 10)
2023-02-14T20:33:16.286Z iteration 0 passed in 711 ms

Node.js: v16.9.1
CCXT v2.7.78

RequestBody:
timestamp=1676407148759&symbol=ETH-230215-1450-C&side=BUY&newOrderRespType=RESULT&type=LIMIT&quantity=1&price=10&timeInForce=GTC&recvWindow=10000&signature=abc123

{
  info: {
    orderId: '4728834411570974720',
    symbol: 'ETH-230215-1450-C',
    price: '10.0',
    quantity: '1.00',
    executedQty: '0.00',
    fee: '0',
    side: 'BUY',
    type: 'LIMIT',
    timeInForce: 'GTC',
    reduceOnly: false,
    postOnly: false,
    createTime: '1676406797645',
    updateTime: '1676406797645',
    status: 'ACCEPTED',
    avgPrice: '0',
    source: 'API',
    clientOrderId: '',
    priceScale: '1',
    quantityScale: '2',
    optionSide: 'CALL',
    quoteAsset: 'USDT',
    mmp: false
  },
  id: '4728834411570974720',
  clientOrderId: undefined,
  timestamp: 1676406797645,
  datetime: '2023-02-14T20:33:17.645Z',
  lastTradeTimestamp: 1676406797645,
  symbol: 'ETH/USDT:USDT-230215-1450-C',
  type: 'limit',
  timeInForce: 'GTC',
  postOnly: false,
  reduceOnly: false,
  side: 'buy',
  price: 10,
  triggerPrice: undefined,
  amount: 1,
  cost: 0,
  average: undefined,
  filled: 0,
  remaining: 1,
  status: 'open',
  fee: { currency: 'USDT', cost: 0, rate: undefined },
  trades: [],
  fees: [ { currency: 'USDT', cost: 0, rate: undefined } ],
  stopPrice: undefined
}
```

Reply:
@carlosmiei If you haven't already you might need to go to the binance website and navigate to the options trading interface to enable it, I think some users were having similar issues when I created some options examples using the implicit API

Reply:
@Dan-krm alright going to try that